### PR TITLE
feat/studio/Home: minutely/hourly/daily usage charts

### PR DESCRIPTION
**Pull request by Ziinc**
This PR allows users to switch between different intervals for their project usage charts.

Notable changes:
- Added testing workarounds for radix dropdown component (see https://github.com/supabase/ui/issues/299 for discussion)
  - stubs missing browser apis in `radix.js` setup file
  - `clickDropdown` helper introduced for firing a radix-specific dropdown click
    - why: monkey patching through a mock to fire the pointer event could give unintended side-effects (e.g. clicking twice) and need extra element checking (complicated)and would be implicit (user doesn't know of implicit workaround).
 - using of `interval=` query param to determine the selected chart state. Defaults to `minutely` as discussed in https://github.com/supabase/infrastructure/issues/2550 

### Pre-merge checks
- [x] add in query param forwarding for `projects/[ref]/log-stats` endpoint, similar what is done for #4536 
- [x] add in LF logic to conditionally change query based on selected `interval` parameter
  - [x] staging
  - [x] prod
- [x] visual check
- [x] feature flag + on/off behaviour verification

### CODE REVIEW

File: studio/components/interfaces/Home/ProjectUsage.tsx [hunk 1]
```diff
@@ -1,43 +1,74 @@
...
 import { get } from 'lib/common/fetch'
 import { API_URL, METRICS, DATE_FORMAT } from 'lib/constants'
 
 const DATETIME_FORMAT = 'MMM D, ha'
-
+const CHART_INTERVALS = [
+  {key: 'minutely', label:'60 minutes', startValue: 1, startUnit: 'hour'},
+  {key: 'hourly', label:'24 hours', startValue: 24, startUnit: 'hour'},
+  {key: 'daily', label:'7 days', startValue: 7, startUnit: 'day'},
+]
```
Comment (milanvanschaik) [inline studio/components/interfaces/Home/ProjectUsage.tsx:28]:
```suggestion
const CHART_INTERVALS = [
  { key: 'minutely', label: '60 minutes', startValue: 1, startUnit: 'hour' },
  { key: 'hourly', label: '24 hours', startValue: 24, startUnit: 'hour' },
  { key: 'daily', label: '7 days', startValue: 7, startUnit: 'day' },
]
```

File: studio/jest.config.js [hunk 2]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
   moduleDirectories: ['<rootDir>', 'node_modules'],
-  setupFiles: ["jest-canvas-mock"],
+  setupFiles: ["jest-canvas-mock", "./tests/setup/radix"],
```
Comment (milanvanschaik) [inline studio/jest.config.js:4]:
```suggestion
  setupFiles: ['jest-canvas-mock', './tests/setup/radix'],
```

**Comment by Ziinc** [comment]
Seems like changing the compile target to `es6` causes many tests to fail, especially when it comes to `await` assertions. 

Will need to adjust the radix setup file to use prototype instead of es6 class feature.

**Comment by Ziinc** [comment]
Currently blocked by https://github.com/supabase/infrastructure/issues/3560 (for local styling and behaviour verification).

**Comment by Ziinc** [comment]
Finally figured it out! 🥳 

Apparently, what was causing the disparity between es5 and es6 compilation targets was that in es5, the radix setup for `ResizeObserver` was executing the callback provided, which apparently in es6 does not cause a silent catastrophic failure and blocking of logic. However, when compiled to es5, it would cause silent failures (presumably due to the callback being executed unexpectedly) 

Also, attempting to extend and create a custom PointerEvent in pure es5 proved tricker than expected, since the Event class is an es6 feature and̋ can't be simply extended using prototype inheritance. JSdom also did not seem to agree that my custom PointerEvent was an actual dom event (as there is an internal EventImpl assertion to verify that it is an event).

**Comment by Ziinc** [comment]
Latest demo

https://user-images.githubusercontent.com/22714384/149520751-8ea1c447-a5e5-455f-9a2b-95589bc56bfd.mov

**Comment by Ziinc** [comment]
Logflare endpoint query has also been tweaked to correctly return 0 values for timestamp intervals where there are no requests, instead of the previous behaviour where intervals wihtout any requests were omitted.

File: studio/tsconfig.test.json [hunk 3]
```diff
@@ -1,6 +1,8 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
+    // minimal target of es6 required for extending Event using es6 class features, see radix setup file (tests/setup/radix.js)
```
Comment (kiwicopple) [inline studio/tsconfig.test.json:4]: This looks like invalid JSON - should it be a JS file?

**Review by kiwicopple — APPROVED** [review]
very nice PR! one comment just as an FYI

File: studio/tsconfig.test.json (hunk 3 above)
Comment (Ziinc) [reply studio/tsconfig.test.json:4]: ah nice catch, ts-jest seems to parse it fine, but i'll remove it anyway.

**Pull request merged**